tests/components/minecraft_server/test_init.py

```diff
@@ -122,26 +124,29 @@ async def test_entry_migration(hass: HomeAssistant) -> None:
 
     # Trigger migration.
     with patch(
-        "mcstatus.server.JavaServer.lookup",
+        "homeassistant.components.minecraft_server.api.MinecraftServer.__init__",
         side_effect=[
-            ValueError,
-            JavaServer(host=TEST_HOST, port=TEST_PORT),
-            JavaServer(host=TEST_HOST, port=TEST_PORT),
+            MinecraftServerAddressError,  # async_migrate_entry
+            None,  # async_migrate_entry
+            None,  # async_setup_entry
         ],
+        return_value=None,
     ), patch(
-        "mcstatus.server.JavaServer.async_status",
-        return_value=TEST_JAVA_STATUS_RESPONSE,
+        "homeassistant.components.minecraft_server.api.MinecraftServer.async_get_data",
+        return_value=TEST_JAVA_DATA,
     ):
         assert await hass.config_entries.async_setup(config_entry_id)
         await hass.async_block_till_done()
 
     # Test migrated config entry.
     config_entry = hass.config_entries.async_get_entry(config_entry_id)
     assert config_entry.unique_id is None
-    assert config_entry.data == {
-        CONF_NAME: DEFAULT_NAME,
-        CONF_ADDRESS: TEST_ADDRESS,
-    }
+    assert config_entry.data[CONF_NAME] == DEFAULT_NAME
+    assert config_entry.data[CONF_ADDRESS] == TEST_ADDRESS
+
+    if CONF_TYPE in config_entry.data:
+        assert config_entry.data[CONF_TYPE] == MinecraftServerType.JAVA_EDITION
+
```

Comment:
```suggestion
    assert config_entry.data == {
        CONF_NAME: DEFAULT_NAME,
        CONF_ADDRESS: TEST_ADDRESS,
    }
```

Revert this change as the migration is never setting `CONF_TYPE`